PowerPointLabs/PowerPointLabs/PasteLab/PasteLabMain.cs

```diff
@@ -237,5 +260,18 @@ public static void PasteToPosition(Models.PowerPointSlide slide, bool clipboardI
                 }
             }
         }
+
+        private static ShapeRange RemovePlaceholders(PowerPointSlide slide, ShapeRange shapes)
```

Comment:
you can create a util method that
* selectShapesWhenTypeMatches
* selectShapesWhenTypeNotMatches